### ISSUE TITLE
-e flag not setting variable on specific distribution

### ISSUE THREAD

**Issue by srashee** [conan-io/conan#5212]
To help us debug your issue please explain:

- [X ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [X ] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X ] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Hi,

I'm currently running Conan Version 1.12.3. I understand that this is a bit behind but i've looked through the releases for anything that may remedy this issue and failed to find anything. The issue is I have the following code in my package method

`if ((os.environ['DEBUG']) == "1"): `
and when calling export-pkg I use 
`conan export-pkg . group/debug -e DEBUG=1`

On my Ubuntu machine running the same conan version this seems to work perfectly fine but on my RHEL7 server DEBUG is never set to 1. I have verified that it's not an env issue by running export DEBUG=1 before running that line and that seems to work perfectly fine.

In summary the -e flag does not work for me on RHEL7 but works for me on Ubuntu with the same conan versions.

**Comment by lasote** [comment]
I cannot reproduce the issue. Could you try to:

- Use a different env var name? `FOO` for example
- Print the contents of `os.environ` in the `package` to see if something could be wrong with the casing? `print(os.environ)`.
- Upgrade conan and verify it still fail.

**Comment by srashee** [comment]
Upgrading conan fixed this issue.
